interfaces/builtin/kubernetes_support.go

```diff
@@ -133,21 +133,32 @@ deny ptrace (trace) peer=unconfined,
 # kubelet calls out to systemd-run for some mounts, but not all of them and not
 # unmounts...
 capability sys_admin,
-mount /var/snap/@{SNAP_NAME}/common/{,**} -> /var/snap/@{SNAP_NAME}/common/{,**},
-mount options=(rw, rshared) -> /var/snap/@{SNAP_NAME}/common/{,**},
+mount /var/snap/@{SNAP_INSTANCE_NAME}/*/{,**} -> /var/snap/@{SNAP_INSTANCE_NAME}/*/{,**},
+mount options=(rw, rshared) -> /var/snap/@{SNAP_INSTANCE_NAME}/*/{,**},
 
 /{,usr/}bin/mount ixr,
 /{,usr/}bin/umount ixr,
 deny /run/mount/utab rw,
-umount /var/snap/@{SNAP_INSTANCE_NAME}/common/**,
+umount /var/snap/@{SNAP_INSTANCE_NAME}/*/**,
 `
 
 const kubernetesSupportConnectedPlugAppArmorKubeletSystemdRun = `
   # kubelet mount rules
   capability sys_admin,
   /{,usr/}bin/mount ixr,
-  mount fstype="tmpfs" tmpfs -> /var/snap/@{SNAP_INSTANCE_NAME}/common/**,
+  mount fstype="tmpfs" tmpfs -> /var/snap/@{SNAP_INSTANCE_NAME}/*/**,
```

Comment:
hmm actually I'm not sure that we want `*` here instead of `common`, wouldn't that allow a snap revision to mount stuff on a different revision? i.e. `mount --bind /var/snap/k8s/4 /var/snap/k8s/10`

I was just thinking something like 
```
  mount fstype="tmpfs" tmpfs -> /var/snap/@{SNAP_INSTANCE_NAME}/{common,@{SNAP_REVISION}}/**,
```

Reply:
Yes, but that was intentional considering 'refresh-mode: skip-refresh'. I've reverted it per IRC conversation (see the revert's commit message).